wsmaster/che-core-api-devfile/src/main/resources/schema/devfile.json

```diff
@@ -94,6 +94,10 @@
                 ]
               }
             }
+          },
+          "clonePath": {
+            "type": "string",
+            "description": "The path relative to the root of the projects to which this project should be cloned into. This is a unix-style relative path (i.e. uses forward slashes). The path is invalid if it is absolute or tries to escape the project root through the usage of '..'. If not specified, defaults to the project name."
```

Comment:
You may define `pattern` to make sure that no `..` nor absolute path is used.

Reply:
`..` can actually be OK. If people end up for example automating devfiles for whatever purpose they might end up needing it, IMHO. You're right about the absolute path though, that can be rejected by the pattern.